xbmc/epg/EpgInfoTag.cpp

```diff
@@ -465,6 +489,144 @@ std::string CEpgInfoTag::PlotOutline(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
+    retVal = m_strIMDBNumber;
+
+  return retVal;
+}
+
+void CEpgInfoTag::SetCast(const std::string &strCast)
+{
+  bool bUpdate(false);
+  {
+    CSingleLock lock(m_critSection);
+    bUpdate = m_strCast != strCast;
+    m_bChanged |= bUpdate;
+    m_strCast = strCast;
+  }
+  if (bUpdate)
+    UpdatePath();
+}
+
+std::string CEpgInfoTag::Cast(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
+    retVal = m_strCast;
+
+  return retVal;
+}
+
+void CEpgInfoTag::SetDirector(const std::string &strDirector)
+{
+  bool bUpdate(false);
+  {
+    CSingleLock lock(m_critSection);
+    bUpdate = m_strDirector != strDirector;
+    m_bChanged |= bUpdate;
+    m_strDirector = strDirector;
+  }
+  if (bUpdate)
+    UpdatePath();
+}
+
+std::string CEpgInfoTag::Director(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
+    retVal = m_strDirector;
+
+  return retVal;
+}
+
+void CEpgInfoTag::SetWriter(const std::string &strWriter)
+{
+  bool bUpdate(false);
+  {
+    CSingleLock lock(m_critSection);
+    bUpdate = m_strWriter != strWriter;
+    m_bChanged |= bUpdate;
+    m_strWriter = strWriter;
+  }
+  if (bUpdate)
+    UpdatePath();
+}
+
+std::string CEpgInfoTag::Writer(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
+    retVal = m_strWriter;
+
+  return retVal;
+}
+
+void CEpgInfoTag::SetOriginalTitle(const std::string &strOriginalTitle)
+{
+  bool bUpdate(false);
+  {
+    CSingleLock lock(m_critSection);
+    bUpdate = m_strOriginalTitle != strOriginalTitle;
+    m_bChanged |= bUpdate;
+    m_strOriginalTitle = strOriginalTitle;
+  }
+  if (bUpdate)
+    UpdatePath();
+}
+
+std::string CEpgInfoTag::OriginalTitle(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
+    retVal = m_strOriginalTitle;
+
+  return retVal;
+}
+
+void CEpgInfoTag::SetYear(const std::string &strYear)
+{
+  bool bUpdate(false);
+  {
+    CSingleLock lock(m_critSection);
+    bUpdate = m_strYear != strYear;
+    m_bChanged |= bUpdate;
+    m_strYear = strYear;
+  }
+  if (bUpdate)
+    UpdatePath();
+}
+
+std::string CEpgInfoTag::Year(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
```

Comment:
you've added parental lock checks for all of these. it's only needed for fields that we don't want kids to see, and I don't think the year or cast are among those things really.